CountLinesExtension/CountLinesExtension.cs

```diff
@@ -24,8 +21,8 @@ public class CountLinesExtension : SharpContextMenu
         /// </returns>
         protected override bool CanShowMenu()
         {
-            //  We always show the menu.
-            return true;
+            //  We only show the menu if any selected file ends with .xml.
+            return SelectedItemPaths?.Any(path => path.EndsWith(".xml")) ?? false;
```

Comment:
This is, why we can use AllFiles - we just filter out all files that don't match.